Trimble.Modus.Components/Controls/TMSwitch/TMSwitch.xaml.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            if (String.IsNullOrEmpty((string)newValue))
+            {
+                tMSwitch.container.Spacing = 0;
+                tMSwitch.switchText.IsVisible = false;
+            }
+            else
+            {
+                tMSwitch.container.Spacing = 8;
+                tMSwitch.switchText.IsVisible = true;
+                tMSwitch.switchText.Text = (string)newValue;
+            }
+
+        }
+    }
+    private static void OnSwitchSizeChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable != null && bindable is TMSwitch tMSwitch)
+        {
+            UpdateSwitchSize(tMSwitch);
+        }
+    }
+    private static void OnSwitchToggleChanged(BindableObject bindable, object oldValue, object newValue)
+    {
+        if (bindable != null && bindable is TMSwitch tMSwitch)
+        {
+            if ((bool)newValue)
+            {
+                tMSwitch._switchSelected = true;
+                OnSwitchSelected(tMSwitch);
+            }
+            else
+            {
+                tMSwitch._switchSelected = false;
+                OnSwitchUnSelected(tMSwitch);
+            }
+        }
+    }
+    private void UpdateSwitch(TMSwitch tMSwitch)
+    {
+
+        tapGestureRecognizer = new TapGestureRecognizer();
+        tapGestureRecognizer.Tapped += OnSwitchTapped;
+        border.GestureRecognizers.Add(tapGestureRecognizer);
+        tMSwitch.border.BackgroundColor = ResourcesDictionary.ColorsDictionary(ColorsConstants.SwitchUnselected);
+        tMSwitch.border.Padding = 5;
+    }
+
+    private static void UpdateSwitchSize(TMSwitch tMSwitch)
+    {
+        tMSwitch.border.Padding = 6;
+        if (tMSwitch.SwitchSize == SwitchSize.Medium)
+        {
+            tMSwitch.border.HeightRequest = 24;
+            tMSwitch.border.WidthRequest = 48;
+            tMSwitch.circle.HeightRequest = 12;
+            tMSwitch.circle.WidthRequest = 12;
+        }
+        else
+        {
+            tMSwitch.border.HeightRequest = 32;
+            tMSwitch.border.WidthRequest = 64;
+            tMSwitch.circle.HeightRequest = 16;
+            tMSwitch.circle.WidthRequest = 16;
+        }
+    }
+
+    private void OnSwitchTapped(object sender, TappedEventArgs e)
+    {
+        _switchSelected = !_switchSelected;
+        if (!_switchSelected)
+        {
+            OnSwitchUnSelected(this);
+        }
+        else
+        {
+            OnSwitchSelected(this);
+        }
+    }
+    private static void OnSwitchSelected(TMSwitch tMSwitch)
+    {
+        tMSwitch._clicked?.Invoke(tMSwitch, tMSwitch.switchEventTrue);
+        tMSwitch.ToggledCommand?.Execute(tMSwitch.switchEventTrue);
+
+        if (tMSwitch.SwitchSize == SwitchSize.Medium)
+        {
+            tMSwitch.Animate("SelectedAnimation",
+                new Animation
+                {
+            { 0, 1, new Animation(v => tMSwitch.circle.Scale = v, 1.0, 1.36) },
```

Comment:
Please explain the magic numbers here. Define them in a variable inside this method itself.
Like `var animationDuration = 250; var something = 1.36`

Reply:
Logic is changed now so no need to define variables